integration_tests/src/main/python/asserts.py

```diff
@@ -558,7 +558,36 @@ def run_on_gpu():
 def run_with_cpu_and_gpu(func,
     mode,
     conf={}):
-    from_cpu, _, from_gpu, _ = run_with_cpu_and_gpu_return_df(func, mode, conf)
```

Comment:
Yes, I added `run_with_cpu_and_gpu_return_df` in this PR to support the new tests. Because it was largely the same as `run_with_cpu_and_gpu`, I thought I would be smart and refactor `run_with_cpu_and_gpu` to call `run_with_cpu_and_gpu_return_df` and then throw away the unwanted values:

```
from_cpu, _, from_gpu, _ = run_with_cpu_and_gpu_return_df(func, mode, conf)
```

However, this was only safe for tests with the mode `COLLECT_WITH_DATAFRAME` because in other cases only one value is returned from `with_cpu_session`.

Reply:
So `run_with_cpu_and_gpu_return_df` is possibly redundant since it should just return the tuple in this case? I will take a look tomorrow.

Reply:
Yeah, this new function was not needed ... I just needed to call `run_with_cpu_and_gpu` with `COLLECT_WITH_DATAFRAME.` I have pushed changes for this.